### ISSUE TITLE
Should make the doc of `nn.CrossEntropyLoss()` more clear

### ISSUE THREAD

**Issue by hyperkai** [pytorch/pytorch#134853]
### 📚 The doc issue

[The doc](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html) of `nn.CrossEntropyLoss()` explains about `target` tensor in a complex way as shown below. *It's difficult to understand:

![Screenshot 2024-08-30 195003](https://github.com/user-attachments/assets/018b6938-14ba-4fab-858a-f99dab4583eb)

So from my understanding and experiments, these simple explanations below should be added to the doc above. *It's easy to understand:

  - The `target` tensor whose size is different from `input` tensor  is treated as class indices.
  - The `target` tensor whose size is same as `input` tensor is the class probabilities which should be between `[0, 1]`.
 
And from what the doc says below and my experiments, when `target` tensor is treated as **class indices**, `softmax()` is used both for `input` and `target` tensor internally:

> The target that this criterion expects should contain either:   
> - Class indices in the range ...     
> ...    
> Note that this case is equivalent to applying [LogSoftmax](https://pytorch.org/docs/stable/generated/torch.nn.LogSoftmax.html#torch.nn.LogSoftmax) on an input, followed by [NLLLoss](https://pytorch.org/docs/stable/generated/torch.nn.NLLLoss.html#torch.nn.NLLLoss).

But when `target` tensor is treated as **class probabilities**, `softmax()` is used only for `input` tensor internally, that's why the example of `target` tensor as **class indices** in the doc doesn't use `softmax()` externally while the example of `target` tensor as **class probabilities** in the doc uses `softmax()` externally as shown below:

![Screenshot 2024-08-30 201043](https://github.com/user-attachments/assets/0f2072f1-0749-45ff-8060-01688680f646)

So, the doc also should say something like as shown below. *You also use the words **class indices mode** and **class probabilities mode** :

- `softmax()` is used internally for `input` tensor, both when `target` tensor is treated as **class indices** and **class probabilities** so you don't need to use `softmax()` externally.
- `softmax()` is used internally for `target` tensor only when `target` tensor is treated as **class indices** so you should use `softmax()` externally for `target` tensor when `target` tensor is treated as **class probabilities**.

cc @svekars @brycebortree @tstatler @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

**Comment by mikaylagawarecki** [comment]
I don't it is the case that softmax is applied on `target` when target is class indices. Please correct me if I missed this though.

Secondly, not sure it is the case that you should use `softmax` externally for the target tensor when target tensor is treated as class probabilities...the word class probabilities implies a probability distribution, softmax a way to generate a probability distribution (and is probably the most common one indeed)

That said, I agree that the docs might not be that intuitive to a lay user and happy to review attempts to improve it
